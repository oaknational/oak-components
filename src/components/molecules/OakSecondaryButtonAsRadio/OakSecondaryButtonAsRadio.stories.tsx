import React from "react";
import { StoryObj, Meta } from "@storybook/react";

import { OakSecondaryButtonAsRadio } from "./OakSecondaryButtonAsRadio";

const meta: Meta<typeof OakSecondaryButtonAsRadio> = {
  component: OakSecondaryButtonAsRadio,
  tags: ["autodocs"],
  argTypes: {},
  parameters: {
    controls: {
      include: ["type"],
    },
  },
};

export default meta;

type Story = StoryObj<typeof OakSecondaryButtonAsRadio>;

export const Default: Story = {
  render: () => (
    <OakSecondaryButtonAsRadio value="option-1">
      Option 1
    </OakSecondaryButtonAsRadio>
  ),
};
