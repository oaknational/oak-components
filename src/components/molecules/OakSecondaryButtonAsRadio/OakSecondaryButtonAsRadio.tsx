import React, { useContext } from "react";

import { OakSecondaryButton } from "@/components/molecules/OakSecondaryButton";
import { RadioContext } from "@/components/molecules/OakButtonAsRadioGroup/OakButtonAsRadioGroup";
import { OakPrimaryButton } from "@/components/molecules/OakPrimaryButton";

export type OakSecondaryButtonAsRadioProps = {
  children?: React.ReactNode;
  value: string;
};

/**
 *
 * To be used as a child of OakButtonAsRadioGroup.
 * Highlights when the value matches the current value of the group.
 * Changes the current value of the group when clicked.
 *
 * ## Usage
 *
 * See OakButtonAsRadioGroup.
 *
 */

export const OakSecondaryButtonAsRadio = ({
  children,
  value,
}: OakSecondaryButtonAsRadioProps) => {
  const radioContext = useContext(RadioContext);
  const { currentValue, onValueUpdated } = radioContext;
  const checked = currentValue === value;

  return checked ? (
    <OakPrimaryButton role="radio" aria-checked={checked}>
      {children}
    </OakPrimaryButton>
  ) : (
    <OakSecondaryButton
      role="radio"
      aria-checked={checked}
      onClick={() => onValueUpdated && onValueUpdated(value)}
    >
      {children}
    </OakSecondaryButton>
  );
};
