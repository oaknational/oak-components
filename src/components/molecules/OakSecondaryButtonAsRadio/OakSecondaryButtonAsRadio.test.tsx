import React from "react";
import "@testing-library/jest-dom";

import { OakSecondaryButtonAsRadio } from "./OakSecondaryButtonAsRadio";

import renderWithTheme from "@/test-helpers/renderWithTheme";

describe("OakSecondaryButtonAsRadio", () => {
  it("matches snapshot", () => {
    const { container } = renderWithTheme(
      <OakSecondaryButtonAsRadio value="1">
        Display Value
      </OakSecondaryButtonAsRadio>,
    );
    expect(container).toMatchSnapshot();
  });
});
