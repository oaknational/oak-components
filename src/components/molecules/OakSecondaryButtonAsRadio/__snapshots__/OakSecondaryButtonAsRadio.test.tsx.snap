// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakSecondaryButtonAsRadio matches snapshot 1`] = `
.c0 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  font-family: Lexend,sans-serif;
}

.c2 {
  position: absolute;
  top: 0rem;
  width: 100%;
  height: 100%;
  border-radius: 0.25rem;
  font-family: Lexend,sans-serif;
}

.c5 {
  font-family: Lexend,sans-serif;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0.5rem;
}

.c7 {
  font-family: Lexend,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
  text-align: left;
}

.c3 {
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  text-align: left;
  font-family: unset;
  outline: none;
  font-family: Lexend,sans-serif;
  color: #222222;
  background: #ffffff;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  border: 0.125rem solid;
  border-color: #222222;
  border-radius: 0.25rem;
}

.c3:disabled {
  pointer-events: none;
  cursor: default;
}

.c4 {
  position: relative;
  width: 100%;
  height: 100%;
  display: inline-block;
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #222222;
  background: #f2f2f2;
  border-color: #222222;
}

.c4:active {
  background: #ffffff;
  border-color: #222222;
  color: #222222;
}

.c4:disabled {
  background: #e4e4e4;
  border-color: #808080;
  color: #808080;
}

.c1 .grey-shadow:has(+ * + .internal-button:focus-visible) {
  box-shadow: 0 0 0 0.3rem rgba(87,87,87,100%);
}

.c1 .yellow-shadow:has(+ .internal-button:focus-visible) {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%);
}

.c1 .yellow-shadow:has(+ .internal-button:hover),
.c1 .yellow-shadow:has(+ .internal-button:hover:not(:focus-visible,:active)) {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%);
}

.c1 .grey-shadow:has(+ * + .internal-button:hover) {
  box-shadow: none;
}

.c1 .grey-shadow:has(+ * + .internal-button:active) {
  box-shadow: 0.25rem 0.25rem 0 rgba(87,87,87,100%);
}

.c1 .yellow-shadow:has(+ .internal-button:active) {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%);
}

<div
  className="c0 c1"
>
  <div
    className="c2 grey-shadow"
  />
  <div
    className="c2 yellow-shadow"
  />
  <button
    aria-checked={false}
    className="c3 c4 internal-button"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="radio"
  >
    <div
      className="c5 c6"
    >
      <span
        className="c7"
      >
        Display Value
      </span>
    </div>
  </button>
</div>
`;
